src/promptflow/promptflow/executor/_async_nodes_scheduler.py

```diff
@@ -12,17 +12,22 @@
 import traceback
 from asyncio import Task
 from concurrent.futures import ThreadPoolExecutor
-from typing import Any, Dict, List, Tuple
+from typing import Any, Dict, List, Optional, Tuple
 
 from promptflow._core.flow_execution_context import FlowExecutionContext
 from promptflow._core.tools_manager import ToolsManager
 from promptflow._utils.logger_utils import flow_logger
 from promptflow._utils.utils import extract_user_frame_summaries, set_context
 from promptflow.contracts.flow import Node
 from promptflow.executor._dag_manager import DAGManager
-from promptflow.executor._errors import NoNodeExecutedError
+from promptflow.executor._errors import LineExecutionTimeoutError, NoNodeExecutedError
 
 PF_ASYNC_NODE_SCHEDULER_EXECUTE_TASK_NAME = "_pf_async_nodes_scheduler.execute"
+PF_ASYNC_NODE_SCHEDULER_EXECUTE_WITH_THREAD_POOL = "_pf_async_nodes_scheduler._execute_with_thread_pool"
```

Comment:
Will the name of scheduler task change after your change? Or your change will introduce another task?

Reply:
The asyncio.wait_for will introduce another task. If we don't set the name of this task, it will raise warning during the process. 
![1706258356022](https://github.com/microsoft/promptflow/assets/39176492/9d607680-f8b5-4afa-867c-2abe55bce423)